dvc/utils/__init__.py

```diff
@@ -349,45 +349,61 @@ def resolve_output(inp, out):
     return out
 
 
-def resolve_paths(repo, out):
+def resolve_paths(repo, out, force_out=False):
```

Comment:
@julsal could you please clarify the essence of this change?

Reply:
The original function receives `repo` and `out`, returning `path` (the `.dvc` file path), `wdir` (the dir where the file should be saved, if `out` is not a path out of `repo`, and a new `out` (usually the `normpath(out)`). 

When `out` is `None`, the name should be inferred from the original, which also impacts `path`. As we don't know the file name at this point  (for the case of `gdrive://ID`), the function now doesn't deal with `path`, and wait to define it into `stage.create_stage`.

One important aspect is that, when `out` is a relative path containing a folder (i.e. `data/file.txt`), `data/` should be incorporated by `wdir` and `out` becomes only `file.txt`. If `out` points to a dir (i.e. `path/to/dir`), the path goes to `wdir` and `out` becomes `None`.

In  `repo.add._create_stages`, `resolve_paths` should always return `out != None`, even when `out` points to a directory. In its default behavior, `resolve_paths` appends the full path to `wdir`. `force_out` guarantee the new `out` keeps the leaf dir. So, if the input is `out=path/to/dir` and `force_out=True`, the new `out` is `dir`, and `path/to` goes to `wdir`.